src/openqaoa-core/openqaoa/problems/bpsp.py

```diff
@@ -0,0 +1,617 @@
+        """
+        return self._car_sequence
+
+    @car_sequence.setter
+    def car_sequence(self, sequence):
+        """
+        Setter for the 'car_sequence' property.
+
+        This method validates and sets a new value for the `_car_sequence` attribute. 
+        The validation ensures:
+        1. Each car ID appears exactly twice in the sequence.
+        2. The range of car IDs is continuous (e.g., for IDs 0, 1, 2, both 0 and 2 cannot appear without 1).
+
+        Parameters:
+        ----------
+        sequence : list[int]
+            A new sequence of car IDs to be assigned to `_car_sequence`. 
+
+        Raises:
+        ------
+        ValueError:
+            If any of the validation checks fail, a ValueError is raised with an appropriate error message.
+
+        """
+        # Check if each car ID appears exactly twice
+        unique_ids, counts = np.unique(sequence, return_counts=True)
+        if not all(count == 2 for count in counts):
+            raise ValueError("Each car ID must appear exactly twice in the sequence.")
+
+        # Check if range of car IDs is continuous
+        if len(unique_ids) != (max(unique_ids) + 1):
+            raise ValueError("The range of car IDs must be continuous.")
+
+        # If all checks pass, assign the sequence to the private attribute
+        self._car_sequence = sequence
+
+    @staticmethod
+    def random_instance(**kwargs):
+        """
+        Generates a random instance of the BPSP problem, based on the specified number of cars.
+        
+        The function creates a list with two occurrences of each car ID, then applies the Fisher-Yates shuffle
+        to randomize the order of the cars. The shuffled sequence of cars is then used to create a BPSP instance.
+        
+        Parameters
+        ----------
+        **kwargs:
+            num_cars: int
+                The number of distinct cars to be considered in the sequence. Each car will appear twice in the 
+                final sequence.
+            seed: int, optional
+                The seed for the random number generator. If provided, the output will be deterministic based on this seed.
+        
+        Returns
+        -------
+            BPSP
+                A random instance of the BPSP problem, based on the shuffled sequence of cars.
+        """
+
+        # Extract the number of cars and the seed from the provided keyword arguments.
+        num_cars = kwargs.get("num_cars")
+        seed = kwargs.get("seed", None)  # Default value is None if not provided.
+
+        # Assert that the number of cars is greater than 0.
+        assert num_cars > 0, "The number of cars must be greater than 0."
+
+        # Generate a list with two occurrences of each car ID, i.e., [0, 1, ..., n, 0, 1, ..., n].
+        car_sequence = np.array(list(range(num_cars)) + list(range(num_cars)))
+
+        # Set the seed for numpy's random module.
+        np.random.seed(seed)
```

Comment:
Use `rng = np.random.default_rng(seed)` and `rng.integers()` to create random numbers

Reply:
Replaced `np.random.seed(seed)` with `rng = np.random.default_rng(seed)` and `rng.integers()` as shown below:

```
# Generate a list with two occurrences of each car ID, i.e., [0, 1, ..., n, 0, 1, ..., n].
        car_sequence = np.array(list(range(num_cars)) + list(range(num_cars)))

        # Create a new instance of the default random number generator.
        rng = np.random.default_rng(seed)

        # Apply the Fisher-Yates shuffle to the car_sequence.
        # Start from the end of the list and swap the current element with a randomly chosen earlier element.
        for i in range(len(car_sequence) - 1, 0, -1):
            # Select a random index between 0 and i (inclusive) using the rng instance.
            j = rng.integers(0, i + 1)
            # Swap the elements at indices i and j.
            car_sequence[i], car_sequence[j] = car_sequence[j], car_sequence[i]
```

If this was what you requested, please resolve this issue. Thanks :)